content/articles/reproducible-research-or-why-docker-is-not-the-answer.md

```diff
@@ -0,0 +1,130 @@
+Title: Why Docker is not the answer to reproducible research, and why Nix may be
+Date: 2015-05-17
+Short_summary: We look at how to do properly reproducible research, and why using Docker doesn't work.
+Category: Dev
+Authors: Tom
+
+
+In this post I'm going to assume that
+[reproducibility](http://en.wikipedia.org/wiki/Reproducibility) is
+good and necessary. There are
+[well-written articles](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003285)
+on how to do reproducible computational research so I don't want to go
+into that either.
+
+Instead I want to focus on how researchers can run published source
+code on their own machines and arrive at the same results.
+
+<!-- PELICAN_END_SUMMARY -->
+
+## The issue with Docker
+
+Unless you are publishing
+[MIX assembly](http://en.wikipedia.org/wiki/MIX) all modern code comes
+with a complex set of dependencies. Many of those dependencies - like
+the GCC compiler - are invisible and taken for granted, but they are
+still dependencies.
+
+Recently I read a few blog posts where people got excited about using
+Docker for reproducibility, and that is a scary idea. What follows is
+a fairly common part of a `Dockerfile`, in this case for the
+[Jupyter notebook](https://github.com/jupyter/notebook/blob/master/Dockerfile):
+
+```
+RUN apt-get update && apt-get install -y language-pack-en
```

Comment:
this is a pretty bad example of the dockerfile, I would copy https://github.com/jupyter/notebook/blob/master/Dockerfile#L22-L42